Local baserow services return human field names

jrmi opened the pull request:
### What is in this PR

To fix the bug mentionned in the issue #4135 all local baserow service need to return the human named data instead of the technical field. This first part was easy to solve. However the data paths in the `get()` formula function still contain the `field_x` and we want to keep that because it makes it more robust when you change the field names.

To solve the second problem, this PR introduces two service methods:
- `get_value_at_path` that returns the value for the given path
- `sanitize_results` that cleans out non allowed field

These two methods can be used to convert the technical names to the human name on the fly.

Lots of test files are updated but most of the changes are only to rename the local baserow service result properties so it should be fine.

### How to test this PR

The first test is to create a workflow with a local baserow trigger and an iterator node. Test both and you should notice that with this branch the schema of the iterator node doesn't contain the `field_x` properties.

I had to refactor the way element content works so it needs extra testing on all collection element especially with nested collection elements and all edge cases. Surprise me on this because I tested so many permutations but I'm sure I've missed something.

You you also test actions inside these elements.

### Checklist

- [ ] A changelog entry has been added to `changelog/entries/unreleased` using `changelog/src/changelog.py`
- [ ] New/updated **Premium/Enterprise features** are separated correctly in the premium or enterprise folder
- [ ] The latest **Chrome and Firefox** have been used to test any new frontend features
- [ ] [Documentation](https://github.com/baserow/baserow/blob/master/docs/index.md) has been updated
- [ ] [Quality Standards](https://github.com/baserow/baserow/blob/master/CONTRIBUTING.md#quality-standards) are met
- [ ] **Performance**: tables are still fast with 100k+ rows, 100+ field tables
- [ ] The [redoc API pages](https://api.baserow.io/api/redoc/) have been updated for any REST API changes
- [ ] Our [custom API docs](https://github.com/baserow/baserow/blob/master/web-frontend/modules/database/pages/APIDocsDatabase.vue) are updated for changes to endpoints accessed via API tokens
- [ ] The UI/UX has been updated following the [UI Style Guide](https://baserow.io/style-guide)
- [ ] Security impact of change has been considered

paljort commented on web-frontend/modules/builder/store/elementContent.js:
```diff
@@ -263,10 +189,10 @@ const actions = {
           // empty array if the property doesn't exist, this will happen if
           // the property has been removed since the initial configuration.
-          const propertyValue = data[element.schema_property] || []
+          // const propertyValue = data[element.schema_property] || []
```
I think we can delete this line 😃

paljort commented on backend/src/baserow/contrib/integrations/core/service_types.py:
```diff
@@ -1609,7 +1610,7 @@ def generate_schema(
         ):
             schema_builder = SchemaBuilder()
-            schema_builder.add_object(service.sample_data["data"])
+            schema_builder.add_object(service.sample_data["data"]["results"])
```
I'm wondering if we need a data migration for this change?

This works perfectly for new iterator nodes, but if there is an existing one (from `develop`), this line causes an exception:
```
TypeError: list indices must be integers or slices, not str
```

In my test, on the `develop` branch I created a Rows Created trigger node with an Iterator node. `service.sample_data` looks like:
```
{
  "data": [
    {
      "id": 1,
      "order": "1.00000000000000000000",
      "field_7257": "Ada1",
      ...
      ]
    }
  ],
  "status": 200,
  "output_uid": ""
}
```

jrmi replied:
It's not needed because we haven't made a release with that version so it's safe to not migrate it.

jrmi commented:
> This is working well and the human friendly names is definitely nicer to read 🚀

Hey @paljort thanks for your review. I fixed everything, would to mind to check the last commit before I merge?

PS: I added a changelog as it affects AB.

paljort reviewed (approved):
Looks great 🚀
